### PR TITLE
Symmterise algorithm: New implementation for finding XMax

### LINKED ISSUE #26179: Indirect Symmetrise fails when EMax=0.4
When trying to symmetrise the data, the algorithm fails if the EMax value is in the range 0.399992 <= EMax <= 0.400090.

### Expected behavior
The data should be symmetrised correctly as it is with any other value of EMax.

### Actual behavior
The algorithm fails with error `Failed to calculate array slice boundaries: index -1 is out of bounds for axis 0 with size 0`

### Steps to reproduce the behavior
- MantidPlot->Interfaces->Indirect->Data Reduction->ISIS Energy transfer
- Set intrument=IRIS, Analyser=Graphite, Reflection=002
- Select a run number (eg. 26176) and press Run
- Data reduction->Symmetrise
- Input Workspace->iris<run number>_graphite002_red
- Set EMax to 0.4
- Press Preview

### Platforms affected
Tested on Ubuntu 18

### PR DESCRIPTION
**Pull request by dtasev**
**Description of work.**
Changes the logic used to find XMax. The -XMin and +XMin logic is unchanged.

**To test:**
Follow instructions from associated issue. Unit tests should pass

Fixes #26179

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: Framework/PythonInterface/plugins/algorithms/Symmetrise.py [hunk 1]
```diff
@@ -241,23 +241,20 @@ def _calculate_array_points(self, sample_x, sample_array_len):
...
-        # Find array index of negative XMin
         negative_min_diff = np.absolute(sample_x + self._x_min)
         self._negative_min_index = np.where(negative_min_diff < delta_x)[0][-1]
         self._check_bounds(self._negative_min_index, sample_array_len, label='Negative')
 
-        # Find array index of positive XMin
+        # Find array index of the first positive XMin, that is smaller than the required
         positive_min_diff = np.absolute(sample_x + sample_x[self._negative_min_index])
         self._positive_min_index = np.where(positive_min_diff < delta_x)[0][-1]
-        self._check_bounds(self._positive_min_index, sample_array_len, label='Positive')
```
Comment (robertapplin) [inline Framework/PythonInterface/plugins/algorithms/Symmetrise.py]: Did you mean to remove this line?

**Review by robertapplin — CHANGES_REQUESTED** [review]
Good job, it works as expected now! Just one minor comment below

File: Framework/PythonInterface/plugins/algorithms/Symmetrise.py (hunk 1 above)
Comment (DTasev) [reply Framework/PythonInterface/plugins/algorithms/Symmetrise.py]: No, good catch!

**Review by robertapplin — APPROVED** [review]
Good job 👍

**Pull request merged**